src/framework/auth/services/token/token.ts

```diff
@@ -121,3 +125,93 @@ export class NbAuthJWTToken extends NbAuthSimpleToken {
     return super.isValid() && (!this.getTokenExpDate() || new Date() < this.getTokenExpDate());
   }
 }
+
+const validateOAuth2Token = (data) => {
+  let result;
+  if (typeof data === 'string') {
+    try {
+      result = JSON.parse(data)
+    } catch (e) {}
+  }
+  return result;
+};
+
+/**
+ * Wrapper for OAuth2 token
+ */
+export class NbAuthOAuth2Token extends NbAuthSimpleToken {
+
+  static NAME = 'nb:auth:oauth2:token';
+
+  constructor(protected data: { [key: string]: string|number }|string = {}) {
+    // we may get it as string when retrieving from a storage
+    super(validateOAuth2Token(data));
+  }
+
+  /**
+   * Returns the token value
+   * @returns string
+   */
+  getValue(): string {
```

Comment:
I don't think the naming of the *getValue* and *getRefresh* is okay. Maybe, it would be better name this methods: *getAccessToken* and *getRefreshToken*?

Reply:
so `token.getAccessToken`? 🤓

Reply:
I think it would be better than *getValue()*